wsmaster/codenvy-hosted-api-audit/src/main/java/com/codenvy/api/audit/server/AuditManager.java

```diff
@@ -0,0 +1,178 @@
+     *
+     * @return path of audit report file
+     * @throws ServerException
+     *         if an error occurs
+     * @throws ConflictException
+     *         if generating report is being run by other user
+     */
+    public Path generateAuditReport() throws ServerException, ConflictException, IOException {
+        if (!inProgress.compareAndSet(false, true)) {
+            throw new ConflictException("This command is being run by other user");
+        }
+
+        Path auditReport = null;
+        try {
+            String dateTime = new SimpleDateFormat("dd-MM-yyyy_hh:mm:ss").format(new Date());
+            auditReport = createTempDirectory(null).resolve("report_" + dateTime + ".txt");
+            Files.createFile(auditReport);
+
+            CodenvyLicense license = null;
+            try {
+                license = licenseManager.load();
+            } catch (LicenseException ignored) {
+                //Continue printing report without license info
+            }
+            reportPrinter.printHeader(auditReport, userManager.getTotalCount(), license);
+            printAllUsers(auditReport);
+        } catch (Exception exception) {
+            if (auditReport != null) {
+                deleteReportDirectory(auditReport);
+            }
+            LOG.error(exception.getMessage(), exception);
+            throw new ServerException(exception.getMessage(), exception);
+        } finally {
+            inProgress.set(false);
+        }
+
+        return auditReport;
+    }
+
+    private void printAllUsers(Path auditReport) throws ServerException {
+        Page<UserImpl> currentPage = userManager.getAll(30, 0);
+        do {
+            //Print users with their workspaces from current page
+            for (UserImpl user : currentPage.getItems()) {
+                List<WorkspaceImpl> workspaces;
+                int permissionsNumber;
+                try {
+                    workspaces = workspaceManager.getWorkspaces(user.getId());
+                    permissionsNumber = workspaces.size();
+                    //add workspaces witch are belong to user, but user doesn't have permissions for them.
+                    workspaceManager.getByNamespace(user.getName())
+                                    .stream()
+                                    .filter(workspace -> !workspaces.contains(workspace))
+                                    .forEach(workspaces::add);
+                } catch (ServerException exception) {
+                    reportPrinter.printError("Failed to receive list of related workspaces for user " + user.getId(), auditReport);
+                    continue;
+                }
+                Map<String, List<AbstractPermissions>> wsPermissions = new HashMap<>();
+                for (WorkspaceImpl workspace : workspaces) {
+                    try {
+                        wsPermissions.put(workspace.getId(), permissionsManager.getByInstance("workspace", workspace.getId()));
```

Comment:
I don't say that you should do it in `AuditReportPrinter`. I just say that you can give `AuditReportPrinter` only one permissions objects instead of `List`